packages/wrangler/src/__tests__/configuration.test.ts

```diff
@@ -6037,6 +6038,57 @@ describe("normalizeAndValidateConfig()", () => {
 	});
 });
 
+describe("experimental_readRawConfig()", () => {
+	describe.each(["json", "jsonc", "toml"])(
+		`with %s config files`,
+		(configType) => {
+			runInTempDir();
+			it(`should find a ${configType} config file given a specific path`, () => {
+				fs.mkdirSync("../folder", { recursive: true });
+				writeWranglerConfig({}, `../folder/config.${configType}`);
+
+				const result = experimental_readRawConfig({
+					config: `../folder/config.${configType}`,
+				});
+				expect(result.rawConfig).toEqual({
+					compatibility_date: "2022-01-12",
+					name: "test-name",
+				});
+			});
+
+			it("should find a config file given a specific script", () => {
+				fs.mkdirSync("./path/to", { recursive: true });
+				writeWranglerConfig(
+					{ name: "config-one" },
+					`./path/wrangler.${configType}`
+				);
+
+				fs.mkdirSync("../folder", { recursive: true });
+				writeWranglerConfig(
+					{ name: "config-two" },
+					`../folder/wrangler.${configType}`
+				);
+
+				let result = experimental_readRawConfig({
+					script: "./path/to/index.js",
+				});
+				expect(result.rawConfig).toEqual({
```

Comment:
nit: should this test `result.rawConfig.name` only? (the date is not specified in the config, same on l 6083.

Reply:
the `writeWranglerConfig` helper provides the default name (`test-name`) and compat date so I think this is okay :)

Reply:
fair enough :) will fix it in a followup PR!